lib/tests-from-pull-request.js

```diff
@@ -7,11 +8,16 @@ function getTestsFromPR(context) {
 
   const repo = context.repo();
   const { pullRequests } = context.github;
+  let config;
 
-  return getPullRequestFiles()
+  return getConfig(context)
+    .then((userConfig) => {
+      config = userConfig;
```

Comment:
In this file, the only attribute you use from `config` is `tdd1tTestFunctionName`.

I think you can save a `testFunctionName` instead in the `getTestsFromPR` function's scope and not save a reference to the `config` itself

Reply:
**mother of all nits:** `getPullRequestFiles();` is unrelated to the "config" function.

I would create a function inside the `getTestsFromPR` scope called `getTestFunctionName` which gets the config and saved it to the scope, while keeping `getPullRequestFiles();` in a separate function in this flow (even though not all the functions in the flow are defined as named functions, but basically the anonymous functions should still be atomic if we can.

Reply:
Do you mean like this:

```
  return getConfig(context)
    .then((userConfig) => getTestFunctionName(userConfig))
    .then((userTestFunctionName) => {
      testFunctionName = userTestFunctionName;
      return getPullRequestFiles();
    })
    .then((files) => files.map((file) => file.patch))
    .then((patches) => patches.join('\n'))
    .then((prPatch) => getTestNames(prPatch, testFunctionName))
    .then((testNames) => {
      context.log.debug({ data: testNames }, 'test names from pull request');

      return testNames;
    });
```

```
function getTestFunctionName(config) {
  return _.get(config, 'tdd1tTestFunctionName', 'it');
}

function getTestName(line, config) {
  const testFunctionName = _.get(config, 'tdd1tTestFunctionName', 'it');
  const getTestNameRegex = new RegExp(`${testFunctionName}\\((.*),`);
  const regexTestResult = getTestNameRegex.exec(line);

  return regexTestResult && regexTestResult[1].trim().slice(1, -1);
}
```

Reply:
I meant like this:
```javascript
function getTestsFromPR(context) {
  let testFunctionName;

  return getTestFunctionName()
    // .then(...)
    // ...

  function getTestFunctionName() {
    return context.config('config.yml')
      .then((config) => _.get(config, 'tdd1tTestFunctionName', 'it'))
      .then((nameFromConfig) => testFunctionName = nameFromConfig);
  }
}
```

or like this:
```javascript
function getTestsFromPR(context) {
  let testFunctionName;

  return getTestFunctionName(context)
    .then((functionName) => testFunctionName = functionName)
    // .then(...)
    // ...
}

function getTestFunctionName(context) {
  return context.config('config.yml')
    .then((config) => _.get(config, 'tdd1tTestFunctionName', 'it'));
}
```

Basically, the first one is defined inside `getTestsFromPR` so **it has access to the context and can set the scope's variable inside the function**, while the second one is defined outside of the function, which means **it needs the context passed into it and to output the functionName**.

Each of them is legit and not too intrusive. The rest of the flow should stay as-is besides passing the `testFunctionName` instead of the `config` along to `getTestNames` & `getTestName`.